### PR TITLE
docs: add docs for spec-strict-refs rule

### PR DESCRIPTION
**Pull request by IgorKarpiuk**
## What/Why/How?

Add docs for spec-strict-refs rule

## Reference
https://github.com/Redocly/redocly-cli/pull/1112
## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: docs/rules/spec-strict-refs.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# spec-strict-refs
+
+Restricts the usage of the `$ref` keyword.
```
Comment (lornajane) [inline docs/rules/spec-strict-refs.md:3]: Let's instead say something like "Check that $ref is only used in the locations permitted by the OpenAPI specification." I think it helps to give a bit more context about why you'd want that.

File: docs/rules/spec-strict-refs.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+|---|---|
+|2.0|✅|
+|3.0|✅|
+|3.1|✅|
+
+## API design principles
+
+This rule is for spec correctness.
+
+Allows to use the `$ref` keyword exclusively for references to elements that may be inside the component section
```
Comment (lornajane) [inline docs/rules/spec-strict-refs.md:15]: Let's also explain that this is useful if you have other tools in your API workflow that require tight compliance with the spec.

**Review by lornajane — CHANGES_REQUESTED** [review]
This looks good, I added a couple of minor comments

File: docs/rules/spec-strict-refs.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+'200':
+  description: successful operation
+  content:
+    application/json:
+      schema:
+        type: object
+        properties:
+          $ref: './properties.yaml'
+          name:
+            type: string
```
Comment (tatomyr) [inline docs/rules/spec-strict-refs.md:77]: You missed the indentation.

File: docs/rules/spec-strict-refs.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+Example of **correct** use of `$ref`:
+
+```yaml Example
+responses:
+'200':
```
Comment (tatomyr) [inline docs/rules/spec-strict-refs.md:84]: The same as above.

File: docs/rules/spec-strict-refs.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+| OAS | Compatibility |
+| --- | ------------- |
+| 2.0 | ✅            |
+| 3.0 | ✅            |
+| 3.1 | ✅            |
+
+## API design principles
+
+This rule is for spec correctness.
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:13]:
```suggestion
This rule ensures adherence to OpenAPI specification standards.
```

File: docs/rules/spec-strict-refs.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+Allows to use the `$ref` keyword exclusively for references to elements that may be inside the component section.
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:15]:
```suggestion
It limits  use of the `$ref` keyword to only references for elements that may be inside the component section.
```

File: docs/rules/spec-strict-refs.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+- RequestBody
+- Header
+- SecurityScheme
+- Link
+- Callback
+- PathItem
+
+## Configuration
+
+To configure the rule, add it to the `rules` object in your configuration file.
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:34]:
```suggestion
To configure the rule, add it to the `rules` object in your configuration file, and
```

File: docs/rules/spec-strict-refs.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+Set the desired [severity](/docs/cli/rules.md#severity-settings) for the rule.
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:35]:
```suggestion
set the desired [severity](/docs/cli/rules.md#severity-settings).
```

File: docs/rules/spec-strict-refs.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+```yaml
+rules:
+  spec-strict-refs: error
+```
+
+## Configuration
+
+| Option   | Type   | Description                              |
+| -------- | ------ | ---------------------------------------- |
+| severity | string | Possible values: `off`, `warn`, `error`. |
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:46]: Should we explain the difference between warn and error? What happens for an error vs. a warning?

File: docs/rules/spec-strict-refs.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+An example configuration:
```
Comment (HCloward) [inline docs/rules/spec-strict-refs.md:48]: Again, not sure this example is necessary since you have a section below with the exact same example.

Comment (IgorKarpiuk) [reply docs/rules/spec-strict-refs.md:41]: These configurations are the same, but they are parts of different sections

**Review by HCloward — APPROVED** [review]

**Review by lornajane — APPROVED** [review]
I think this is ready to go. Thanks @IgorKarpiuk !

**Pull request merged**